applications/nrf5340_audio/src/bluetooth/bt_stream/bt_le_audio_tx/bt_le_audio_tx.c

```diff
@@ -22,44 +23,59 @@ ZBUS_CHAN_DEFINE(sdu_ref_chan, struct sdu_ref_msg, NULL, NULL, ZBUS_OBSERVERS_EM
 
 #define HANDLE_INVALID 0xFFFF
 
-#define HCI_ISO_BUF_ALLOC_PER_CHAN 2
+#define HCI_ISO_BUF_PER_CHAN 2
+
+#if defined(CONFIG_BT_ISO_MAX_CIG) && defined(CONFIG_BT_ISO_MAX_BIG)
+#define GROUP_MAX (CONFIG_BT_ISO_MAX_BIG + CONFIG_BT_ISO_MAX_CIG)
+#elif defined(CONFIG_BT_ISO_MAX_CIG)
+#define GROUP_MAX CONFIG_BT_ISO_MAX_CIG
+#elif defined(CONFIG_BT_ISO_MAX_BIG)
+#define GROUP_MAX CONFIG_BT_ISO_MAX_BIG
+#else
+#error Neither CIG or BIG defined
+#endif
+
+#if defined(CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT)
+#define SUBGROUP_MAX (1 + CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT)
+#else
+/* 1 since CIGs doesn't have the concept of subgroups */
+#define SUBGROUP_MAX 1
+#endif
+
+/* Since we can't assume that the number of streams are equally distributed on the subgroups ,we
+ * need to allocate the max number per subgroup
+ */
+#define TX_STREAMS_MAX (CONFIG_BT_ISO_MAX_CHAN)
+
+#define NET_BUF_POOL_MAX ((TX_STREAMS_MAX) * (SUBGROUP_MAX) * (GROUP_MAX) * (HCI_ISO_BUF_PER_CHAN))
 
-/* For being able to dynamically define iso_tx_pools */
-#define NET_BUF_POOL_ITERATE(i, _)                                                                 \
-	NET_BUF_POOL_FIXED_DEFINE(iso_tx_pool_##i, HCI_ISO_BUF_ALLOC_PER_CHAN,                     \
-				  BT_ISO_SDU_BUF_SIZE(CONFIG_BT_ISO_TX_MTU), 8, NULL);
-#define NET_BUF_POOL_PTR_ITERATE(i, ...) IDENTITY(&iso_tx_pool_##i)
-LISTIFY(CONFIG_BT_ISO_MAX_CHAN, NET_BUF_POOL_ITERATE, (;))
-/* clang-format off */
-static struct net_buf_pool *iso_tx_pools[] = { LISTIFY(CONFIG_BT_ISO_MAX_CHAN,
-						       NET_BUF_POOL_PTR_ITERATE, (,)) };
-/* clang-format on */
+NET_BUF_POOL_FIXED_DEFINE(iso_tx_pool, NET_BUF_POOL_MAX, BT_ISO_SDU_BUF_SIZE(CONFIG_BT_ISO_TX_MTU),
+			  8, NULL);
 
 struct tx_inf {
 	uint16_t iso_conn_handle;
 	struct bt_iso_tx_info iso_tx;
 	struct bt_iso_tx_info iso_tx_readback;
-	struct net_buf_pool *iso_tx_pool;
 	atomic_t iso_tx_pool_alloc;
 	bool hci_wrn_printed;
 };
 
 static bool initialized;
-static struct tx_inf tx_info_arr[CONFIG_BT_ISO_MAX_CHAN];
+static struct tx_inf tx_info_arr[GROUP_MAX][SUBGROUP_MAX][TX_STREAMS_MAX];
```

Comment:
Discuss: Should we rename to LVL1_MAX, LVL2_MAX, LVL3_MAX?

Reply:
We could, but then it's hard to know what those represent if you are not already in the know